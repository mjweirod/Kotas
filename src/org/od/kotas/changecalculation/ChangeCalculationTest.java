package org.od.kotas.changecalculation;
import static org.junit.Assert.*;

import org.approvaltests.Approvals;
import org.approvaltests.legacycode.Range;
import org.junit.Test;


public class ChangeCalculationTest {

	@Test
	public void testAll() throws Exception {
 Approvals.verifyAll(Range.get(1, 99),  i -> String.format("%s -> %s", i,ChangeCalucator.dispense(i) ));
	}
	@Test
	public void testSimpleChange() throws Exception {
		Change change = ChangeCalucator.dispense(1);
		assertEquals(1, change.pennies);
	}
	
	@Test
	public void test2CentsChange() throws Exception {
		Change change = ChangeCalucator.dispense(2);
		assertEquals(2, change.pennies);
	}
	
	@Test
	public void testNickel() throws Exception {
		Change change = ChangeCalucator.dispense(5);
		assertEquals(1,  change.nickels);
	}
}
