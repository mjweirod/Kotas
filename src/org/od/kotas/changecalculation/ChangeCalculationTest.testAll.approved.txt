1 -> Change [pennies=1, nickels=0, dimes=0, quarters=0]
2 -> Change [pennies=2, nickels=0, dimes=0, quarters=0]
3 -> Change [pennies=3, nickels=0, dimes=0, quarters=0]
4 -> Change [pennies=4, nickels=0, dimes=0, quarters=0]
5 -> Change [pennies=0, nickels=1, dimes=0, quarters=0]
6 -> Change [pennies=1, nickels=1, dimes=0, quarters=0]
7 -> Change [pennies=2, nickels=1, dimes=0, quarters=0]
8 -> Change [pennies=3, nickels=1, dimes=0, quarters=0]
9 -> Change [pennies=4, nickels=1, dimes=0, quarters=0]
10 -> Change [pennies=0, nickels=0, dimes=1, quarters=0]
11 -> Change [pennies=1, nickels=2, dimes=0, quarters=0]
12 -> Change [pennies=2, nickels=2, dimes=0, quarters=0]
13 -> Change [pennies=3, nickels=2, dimes=0, quarters=0]
14 -> Change [pennies=4, nickels=2, dimes=0, quarters=0]
15 -> Change [pennies=0, nickels=3, dimes=0, quarters=0]
16 -> Change [pennies=1, nickels=3, dimes=0, quarters=0]
17 -> Change [pennies=2, nickels=3, dimes=0, quarters=0]
18 -> Change [pennies=3, nickels=3, dimes=0, quarters=0]
19 -> Change [pennies=4, nickels=3, dimes=0, quarters=0]
20 -> Change [pennies=0, nickels=4, dimes=0, quarters=0]
21 -> Change [pennies=1, nickels=4, dimes=0, quarters=0]
22 -> Change [pennies=2, nickels=4, dimes=0, quarters=0]
23 -> Change [pennies=3, nickels=4, dimes=0, quarters=0]
24 -> Change [pennies=4, nickels=4, dimes=0, quarters=0]
25 -> Change [pennies=0, nickels=5, dimes=0, quarters=0]
26 -> Change [pennies=1, nickels=5, dimes=0, quarters=0]
27 -> Change [pennies=2, nickels=5, dimes=0, quarters=0]
28 -> Change [pennies=3, nickels=5, dimes=0, quarters=0]
29 -> Change [pennies=4, nickels=5, dimes=0, quarters=0]
30 -> Change [pennies=0, nickels=6, dimes=0, quarters=0]
31 -> Change [pennies=1, nickels=6, dimes=0, quarters=0]
32 -> Change [pennies=2, nickels=6, dimes=0, quarters=0]
33 -> Change [pennies=3, nickels=6, dimes=0, quarters=0]
34 -> Change [pennies=4, nickels=6, dimes=0, quarters=0]
35 -> Change [pennies=0, nickels=7, dimes=0, quarters=0]
36 -> Change [pennies=1, nickels=7, dimes=0, quarters=0]
37 -> Change [pennies=2, nickels=7, dimes=0, quarters=0]
38 -> Change [pennies=3, nickels=7, dimes=0, quarters=0]
39 -> Change [pennies=4, nickels=7, dimes=0, quarters=0]
40 -> Change [pennies=0, nickels=8, dimes=0, quarters=0]
41 -> Change [pennies=1, nickels=8, dimes=0, quarters=0]
42 -> Change [pennies=2, nickels=8, dimes=0, quarters=0]
43 -> Change [pennies=3, nickels=8, dimes=0, quarters=0]
44 -> Change [pennies=4, nickels=8, dimes=0, quarters=0]
45 -> Change [pennies=0, nickels=9, dimes=0, quarters=0]
46 -> Change [pennies=1, nickels=9, dimes=0, quarters=0]
47 -> Change [pennies=2, nickels=9, dimes=0, quarters=0]
48 -> Change [pennies=3, nickels=9, dimes=0, quarters=0]
49 -> Change [pennies=4, nickels=9, dimes=0, quarters=0]
50 -> Change [pennies=0, nickels=10, dimes=0, quarters=0]
51 -> Change [pennies=1, nickels=10, dimes=0, quarters=0]
52 -> Change [pennies=2, nickels=10, dimes=0, quarters=0]
53 -> Change [pennies=3, nickels=10, dimes=0, quarters=0]
54 -> Change [pennies=4, nickels=10, dimes=0, quarters=0]
55 -> Change [pennies=0, nickels=11, dimes=0, quarters=0]
56 -> Change [pennies=1, nickels=11, dimes=0, quarters=0]
57 -> Change [pennies=2, nickels=11, dimes=0, quarters=0]
58 -> Change [pennies=3, nickels=11, dimes=0, quarters=0]
59 -> Change [pennies=4, nickels=11, dimes=0, quarters=0]
60 -> Change [pennies=0, nickels=12, dimes=0, quarters=0]
61 -> Change [pennies=1, nickels=12, dimes=0, quarters=0]
62 -> Change [pennies=2, nickels=12, dimes=0, quarters=0]
63 -> Change [pennies=3, nickels=12, dimes=0, quarters=0]
64 -> Change [pennies=4, nickels=12, dimes=0, quarters=0]
65 -> Change [pennies=0, nickels=13, dimes=0, quarters=0]
66 -> Change [pennies=1, nickels=13, dimes=0, quarters=0]
67 -> Change [pennies=2, nickels=13, dimes=0, quarters=0]
68 -> Change [pennies=3, nickels=13, dimes=0, quarters=0]
69 -> Change [pennies=4, nickels=13, dimes=0, quarters=0]
70 -> Change [pennies=0, nickels=14, dimes=0, quarters=0]
71 -> Change [pennies=1, nickels=14, dimes=0, quarters=0]
72 -> Change [pennies=2, nickels=14, dimes=0, quarters=0]
73 -> Change [pennies=3, nickels=14, dimes=0, quarters=0]
74 -> Change [pennies=4, nickels=14, dimes=0, quarters=0]
75 -> Change [pennies=0, nickels=15, dimes=0, quarters=0]
76 -> Change [pennies=1, nickels=15, dimes=0, quarters=0]
77 -> Change [pennies=2, nickels=15, dimes=0, quarters=0]
78 -> Change [pennies=3, nickels=15, dimes=0, quarters=0]
79 -> Change [pennies=4, nickels=15, dimes=0, quarters=0]
80 -> Change [pennies=0, nickels=16, dimes=0, quarters=0]
81 -> Change [pennies=1, nickels=16, dimes=0, quarters=0]
82 -> Change [pennies=2, nickels=16, dimes=0, quarters=0]
83 -> Change [pennies=3, nickels=16, dimes=0, quarters=0]
84 -> Change [pennies=4, nickels=16, dimes=0, quarters=0]
85 -> Change [pennies=0, nickels=17, dimes=0, quarters=0]
86 -> Change [pennies=1, nickels=17, dimes=0, quarters=0]
87 -> Change [pennies=2, nickels=17, dimes=0, quarters=0]
88 -> Change [pennies=3, nickels=17, dimes=0, quarters=0]
89 -> Change [pennies=4, nickels=17, dimes=0, quarters=0]
90 -> Change [pennies=0, nickels=18, dimes=0, quarters=0]
91 -> Change [pennies=1, nickels=18, dimes=0, quarters=0]
92 -> Change [pennies=2, nickels=18, dimes=0, quarters=0]
93 -> Change [pennies=3, nickels=18, dimes=0, quarters=0]
94 -> Change [pennies=4, nickels=18, dimes=0, quarters=0]
95 -> Change [pennies=0, nickels=19, dimes=0, quarters=0]
96 -> Change [pennies=1, nickels=19, dimes=0, quarters=0]
97 -> Change [pennies=2, nickels=19, dimes=0, quarters=0]
98 -> Change [pennies=3, nickels=19, dimes=0, quarters=0]
99 -> Change [pennies=4, nickels=19, dimes=0, quarters=0]
